### PR TITLE
fix(es/codegen): Fix numeric literals

### PR DESCRIPTION
**Pull request by kdy1**
**Description:**

**Related issue (if exists):**

 - Closes https://github.com/swc-project/swc/issues/4724

### CODE REVIEW

**Review by kdy1 — COMMENTED** [review]
swc-bump:
 - swc_ecma_codegen

File: crates/swc/tests/tsc-references/parser.numericSeparators.decimal_es2015.1.normal.js [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-1_2.3_4;
-1_2.3_4e5_6;
-1_2.3_4e+5_6;
-1_2.3_4e-5_6;
+1000000000;
+1.10001;
+10000000000;
+10000000000;
+0.0000000001;
+11000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000;
```
Comment (magic-akari) [inline crates/swc/tests/tsc-references/parser.numericSeparators.decimal_es2015.1.normal.js:6]: I prefer to keep changes to a minimum.
Maybe it's better to just remove the numeric separator `_`.

```diff
- 11000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000;
+ 1.1e100;
```

File: crates/swc_ecma_codegen/tests/fixture/number/output.js [hunk 2]
```diff
@@ -105,7 +105,7 @@ const foo71 = 0XfF;
 77.0.toExponential();
 const hugefoo = 1000000000001..test();
 const hugefoo1 = 0xEE.test();
-1_2_3_4..toString();
+1234.toString();
```
Comment (magic-akari) [inline crates/swc_ecma_codegen/tests/fixture/number/output.js:108]: It seems like a regression.

Comment (kdy1) [reply crates/swc_ecma_codegen/tests/fixture/number/output.js:108]: Oh you are right

**Pull request merged**